services/director-v2/src/simcore_service_director_v2/modules/dynamic_sidecar/docker_compose_specs.py

```diff
@@ -104,7 +126,92 @@ def _replace_env_vars_in_compose_spec(
     return stringified_service_spec
 
 
-def assemble_spec(
+async def _inject_resource_limits_and_reservations(
+    app: FastAPI, service_key: str, service_tag: str, service_spec: ComposeSpecLabel
+) -> None:
+    director_v0_client: DirectorV0Client = get_director_v0_client(app)
+    labels_for_involved_services: Dict[
+        str, SimcoreServiceLabels
+    ] = await get_labels_for_involved_services(
+        director_v0_client=director_v0_client,
+        service_key=service_key,
+        service_tag=service_tag,
+    )
+
+    # example: '2.3' -> 2 ; '3.7' -> 3
+    docker_compose_major_version: int = int(service_spec["version"].split(".")[0])
+
+    for spec_service_key, spec in service_spec["services"].items():
+        if spec_service_key not in labels_for_involved_services:
+            logger.info(
+                "No labels found for service %s, service may not be an osparc service",
```

Comment:
I am not sure I understand this if case here. so we can send any service in there? should it not fail the process instead?

Reply:
When you bundle a docker-compose spec in the labels of a service, you can specify more than one container for it.
For example you could add a `caddy`. Since `caddy` is not an Osparc service (a.k.a. not present in the service catalog, requesting labels for the service will fail).
That's why no fail is require here.

Reply:
ok, then I have a security question. what prevents me from putting a bad service? how is this checked?